modules/virt-creating-configmap.adoc

```diff
@@ -0,0 +1,113 @@
+// Module included in the following assemblies:
+// * virt/virtual_machines/importing_vms/virt-importing-rhv-vm.adoc
+[id='virt-creating-configmap_{context}']
+= Creating a custom operating system ConfigMap
+
+If the operating system of the Red Hat Virtualization virtual machine does not exist in the default ConfigMap of the `vm-import-controller` Custom Resource (CR), you can create a custom ConfigMap to map the operating system to a common template. The template is used to create the new virtual machine in {VirtProductName}.
+
+.Procedure
+
+. Create a custom ConfigMap:
++
+----
+$ cat <<EOF | oc create -f -
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: os-configmap
+  namespace: default <1>
+data:
+  guestos2common: | <2>
+    "Red Hat Enterprise Linux Server": "rhel"
+    "CentOS Linux": "centos"
+    "Fedora": "fedora"
+    "Ubuntu": "ubuntu"
+    "openSUSE": "opensuse"
+  osinfo2common: | <3>
+    "rhel_6_9_plus_ppc64": "rhel6.9"
+    "rhel_6_ppc64": "rhel6.9"
+    "rhel_6": "rhel6.9"
+    "rhel_6x64": "rhel6.9"
+    "rhel_7_ppc64": "rhel7.7"
+    "rhel_7_s390x": "rhel7.7"
+    "rhel_7x64": "rhel7.7"
+    "rhel_8x64": "rhel8.1"
+    "sles_11_ppc64": "opensuse15.0"
+    "sles_11": "opensuse15.0"
+    "sles_12_s390x": "opensuse15.0"
+    "ubuntu_12_04": "ubuntu18.04"
+    "ubuntu_12_10": "ubuntu18.04"
+    "ubuntu_13_04": "ubuntu18.04"
+    "ubuntu_13_10": "ubuntu18.04"
+    "ubuntu_14_04_ppc64": "ubuntu18.04"
+    "ubuntu_14_04": "ubuntu18.04"
+    "ubuntu_16_04_s390x": "ubuntu18.04"
+    "windows_10": "win10"
+    "windows_10x64": "win10"
+    "windows_2003": "win10"
+    "windows_2003x64": "win10"
+    "windows_2008R2x64": "win2k8"
+    "windows_2008": "win2k8"
+    "windows_2008x64": "win2k8"
+    "windows_2012R2x64": "win2k12r2"
+    "windows_2012x64": "win2k12r2"
+    "windows_2016x64": "win2k16"
+    "windows_2019x64": "win2k19"
+    "windows_7": "win10"
+    "windows_7x64": "win10"
+    "windows_8": "win10"
+    "windows_8x64": "win10"
+    "windows_xp": "win10"
+EOF
+----
+<1> Optional: You can update the namespace.
+<2> `guestos2common` maps the operating system reported by the guest agent to a common template.
+<3> `osinfo2common` maps the operating system resource of the source provider to a common template.
```

Comment:
The values for this map could be obtained by the following commands:

For the mapping of `osinfo2common` (from os-info to common template), the source values on the right are fetched from the oslibinfo - the default values can be found as part of ovirt on https://github.com/oVirt/ovirt-engine/blob/master/packaging/conf/osinfo-defaults.properties or on installed RHV/oVirt under `etc/ovirt-engine/osinfo.conf.d/00-defaults.properties`.

This is a snippet of the file:
```
# Windows2016x64(29, OsType.Windows, true);
os.windows_2016x64.id.value = 29
os.windows_2016x64.name.value = Windows 2016 x64
os.windows_2016x64.derivedFrom.value = windows_2012R2x64
os.windows_2016x64.cpu.unsupported.value = conroe, penryn, nehalem, opteron_g1
os.windows_2016x64.sysprepPath.value = ${ENGINE_USR}/conf/sysprep/sysprep.2k16x64
```
Base on it, the key in the `osinfo2common` map will be `windows_2016x64`.

For the common-template values on the right side of `osinfo2common` map - supported values can be retrieved by the command:
```
oc get templates -n openshift  --show-labels | tr ',' '\n' | grep os.template.kubevirt.io | sed -r 's#os.template.kubevirt.io/(.*)=.*#\1#g' | sort -u
```
which will return the list of all the common-templates OS types installed on OCP cluster.
Here is part of the output for that command:
```
rhel8.0
rhel8.1
rhel8.2

win10
win2k12r2
win2k16
```

For the other mapping named `guestos2common`, source values are taken from the guest-agent tool installed on the VM and reported to ovirt-engine. This map should be stable mapping and less customized.

Another comment about the example - we'd like to emphasis that the user doesn't have to provide the entire mapping, but it is enough to provide only a subset of the values or only one of the mapping (either `guestos2common` or `osinfo2common` ). 

@istein1 pls let me know if that covers the points we talked on.

Reply:
I thought it might be good to turn the OCP command for getting templates to the first step.
I added the RHVM location for the osinfo and a phrase about using only the values appropriate for the VM being imported.